src/services/organizationService.ts

```diff
@@ -140,6 +142,130 @@ function mergePublicAdministrationsAndOrganizations(
+ *  see https://www.pivotaltracker.com/story/show/169889085
+ * @param organizationInstance The instance of the organization to be deleted.
+ */
+export async function deleteOrganization(
+  organizationInstance: OrganizationModel
+): Promise<Option<Error>> {
+  try {
+    await sequelize.transaction(transaction => {
+      return OrganizationUserModel.destroy({
+        force: true,
+        transaction,
+        where: { organizationIpaCode: organizationInstance.ipaCode }
+      })
+        .then(() => {
+          return organizationInstance.destroy({ force: true, transaction });
+        })
+        .then(() => {
+          return organizationInstance.legalRepresentative.destroy({
+            force: true,
+            transaction
+          });
+        });
+    });
+    // TODO:
+    //  the documents must be stored on cloud (Azure Blob Storage).
+    //  @see https://www.pivotaltracker.com/story/show/169644958
+    const organizationDocumentsRoot = `./documents/${organizationInstance.ipaCode}`;
+    await fs.promises
+      .access(organizationDocumentsRoot)
+      .then(() =>
+        fs.promises
+          .readdir(organizationDocumentsRoot)
+          .then(documentsArray =>
+            Promise.all(
+              documentsArray.map(documentName =>
+                fs.promises.unlink(
+                  `${organizationDocumentsRoot}/${documentName}`
+                )
+              )
+            ).then(() => fs.promises.rmdir(organizationDocumentsRoot))
+          )
+      )
+      .catch(error => {
+        log.error(
+          "An error occurred deleting the documents of a deleted organization. %s",
+          error
+        );
+      });
+    return none;
+  } catch (error) {
+    return some(error);
+  }
+}
+
+/**
+ * Checks if the registration process for a given administration has already started
+ * and deletes it if it's still in a pre-draft status. Returns an optional error response
+ * if the user can not start a new registration process for the given administration.
+ * @param ipaCode The ipaCode of the administration to be checked
+ */
+async function checkAndDeletePreviousRegistration(
+  ipaCode: string
+): Promise<Option<IResponseErrorConflict | IResponseErrorInternal>> {
+  try {
+    const oldOrganizationInstance = await OrganizationModel.findOne({
+      include: [
+        {
+          as: "users",
+          model: User,
+          required: true
+        },
+        {
+          as: "legalRepresentative",
+          model: User
+        }
+      ],
+      where: { ipaCode }
+    });
+    if (oldOrganizationInstance !== null) {
+      if (
+        oldOrganizationInstance.registrationStatus !==
+        OrganizationRegistrationStatusEnum.PRE_DRAFT
+      ) {
+        return some(
+          ResponseErrorConflict("The organization is already registered")
+        );
+      }
+      const maybeError = await deleteOrganization(oldOrganizationInstance);
```

Comment:
a better (and safer) way to express this condition is to have a type (ie. `OrganizationDraft`) with 
`registrationStatus: t.literal(OrganizationRegistrationStatusEnum.PRE_DRAFT)` so you can define
`deleteOrganization(o: OrganizationDraft)` to accept _only_ organizations which have a draft status.

Finally, instead of the if condition here you would have 
```ts
if (OrganizationDraft.is(oldOrganizationInstance)) {
   ... deleteOrganization(oldOrganizationInstance) ...
}
``` 
(otherwise you cannot call `deleteOrganization(oldOrganizationInstance)`)

Generally speaking every **condition on object properties** should be defined with a corresponding type. This let you pass around correct objects into functions that expects some object with one ore more condition applied on its fields.